### PR TITLE
CURL: training updates and single-image inference

### PR DESCRIPTION
**Pull request by sjmoran**
This update refines CURL's curve training and evaluation, and adds tests, single-image inference, and a retrained checkpoint. I am one of the paper's authors. All changes are under `CURL/`.

### Changes

- Keep predicted curve knots and slopes as tensors so curve training uses both the image loss and the smoothness regulariser. A dedicated test checks image-loss gradients independently of regularisation.
- Evaluate every curve segment using clamped, piecewise-linear interpolation in a vectorised pass. Use consistent handling of tied RGB maxima during HSV conversion, apply the MS-SSIM top-scale term once, and use a fixed SSIM data range.
- Split model, colour conversion, loss, training, and inference code into focused modules, retaining existing import names.
- Add `pip install -e .` and `curl-enhance enhance photo.jpg`, automatic device selection, and support for common image formats.
- Add configurable regularisation, optional cosine learning-rate scheduling, curve logging and plots, and an updated README with enhancement examples.

### Checkpoint

The supplied checkpoint achieves **24.15 dB PSNR / 0.915 SSIM** on the DPE split. It was trained for 500 epochs with `--reg_weight=1e-2` and `--lr_schedule=cosine`; epoch 270 was selected using validation PSNR. A weights-only copy is included for inference.

This is one training run, without multiple seeds or per-change ablations. Results should only be compared with evaluations using the same DPE protocol.

### Validation

Reported validation on CPU and Apple Silicon with Python 3.11, PyTorch 2.12, scikit-image 0.26, and NumPy 2.4:

- **42 tests passed**, with none skipped, including the curve-gradient test.
- Both supplied checkpoint files loaded without missing or unexpected state-dict keys.
- Editable installation and single-image inference succeeded.
- The 13 bundled examples scored **29.43 dB / 0.957 SSIM**; this example-set result is separate from the DPE benchmark above.

To run the tests without a dataset:

```bash
cd CURL
pip install -r requirements.txt
python -m pytest test_curl.py
```

The curve and loss updates affect training and output behaviour as well as code organisation. CUDA execution is not covered by the reported validation.

### CODE REVIEW

**Comment by sjmoran** [comment]
Hi @huawei-noah-admin, just following up on this CURL update. I noticed this account has handled recent merges in the repository, so I hoped you could help route this review.

I am one of the paper’s authors and would be happy to address any questions or requested changes. Could you review this PR or point me to the current CURL maintainer?

Thanks!
